From 5062fa56f66c295394595808094742c63c352d19 Mon Sep 17 00:00:00 2001
From: Oleg Pudeyev <oleg@bsdpower.com>
Date: Mon, 4 Jun 2018 09:09:17 -0400
Subject: [PATCH] Replace error with a warning for undetected SSL backend.

We don't detect nor do anything special for a multitude of
SSL backends like DarwinSSL, WinSSL and the more obscure ones
available on Unices.

diff --git setup.py setup.py
index 9f124473..06f26c92 100644
--- setup.py
+++ setup.py
@@ -311,10 +311,11 @@ def configure_unix(self):
             self.detect_ssl_backend()
 
             if not self.ssl_lib_detected:
-                raise ConfigurationError('''\
-Curl is configured to use SSL, but we have not been able to determine \
-which SSL backend it is using. Please see PycURL documentation for how to \
-specify the SSL backend manually.''')
+                sys.stderr.write('''\
+Warning: libcurl is configured to use SSL, but we have not been able to \
+determine which SSL backend it is using. If your Curl is built against \
+OpenSSL, LibreSSL, BoringSSL, GnuTLS or NSS please specify the SSL backend \
+manually. For other SSL backends please ignore this message.''')
         else:
             if self.detect_ssl_option():
                 sys.stderr.write("Warning: SSL backend specified manually but libcurl does not use SSL\n")
